pennylane_lightning/lightning_tensor/lightning_tensor.py

```diff
@@ -374,8 +380,11 @@ def _setup_execution_config(
 
         updated_values = {}
 
+        _device_options = (
+            self._device_options_mps if self.method == "mps" else self._device_options_tn
+        )
         new_device_options = dict(config.device_options)
-        for option in self._device_options:
+        for option in _device_options:
```

Comment:
You can even forget about defining _device_options, and just do:
```suggestion
        for option in _device_options[self.method] :
```